src/test/scala/scorex/core/network/peer/PeerManagerSpec.scala

```diff
@@ -83,8 +83,8 @@ class PeerManagerSpec extends FlatSpec with Matchers with ObjectGenerators {
     val data1 = pr1.expectMsgClass(classOf[Data])
     data1.keySet should contain(pa1)
 
-    val h2 = Handshake("test", Version(1: Byte, 2: Byte, 3: Byte), "1", Some(pa2), feats, System.currentTimeMillis())
-    //connected peer became non-public cause declared address != peerAddress
+    val h2 = Handshake("test", Version(1: Byte, 2: Byte, 3: Byte), "1", None, feats, System.currentTimeMillis())
```

Comment:
Does declared address feature works now?

Reply:
I think it works

Reply:
I traied to run HybridApp with changes in this PR.

Nodes with default settings not discover peers for the following reasons: 
1) setting declaredAddress is not defined
2) upnp is disabled

When declaredAdress is definded peer discovery is work fine.